packages-exp/auth-exp/src/core/providers/oauth.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+
+import { AuthErrorCode } from '../errors';
+import { debugFail } from '../util/assert';
+
+export interface CustomParameters {
+  [key: string]: string;
+}
+
+interface CredentialParameters {
+  idToken?: string;
+  accessToken?: string;
+  rawNonce?: string;
+}
+
+export class OAuthProvider implements externs.AuthProvider {
+  defaultLanguageCode: string | null = null;
+  private scopes: string[] = [];
+  private customParameters: CustomParameters = {};
+  constructor(readonly providerId: externs.ProviderId) {}
```

Comment:
make constructor private pls

Reply:
It's meant to be public: https://firebase.google.com/docs/reference/js/firebase.auth.OAuthProvider#constructor